### PR TITLE
Cache control on api

### PR DESCRIPTION
**Pull request by jonasfj**
We should probably add more cache-control on other API end-points that are frequently scanned by bots. Or just in general on everything that can reasonably be cached.

### CODE REVIEW

File: app/lib/shared/handler_helpers.dart [hunk 1]
```diff
@@ -136,7 +136,7 @@ shelf.Handler _requestContextWrapper(shelf.Handler handler) {
     if (!context.uiCacheEnabled && !CacheHeaders.hasCacheHeader(rs.headers)) {
```
Comment (jonasfj) [inline app/lib/shared/handler_helpers.dart:136]: This and the `requestContext.isNotAuthenticated` logic in app/lib/frontend/handlers/headers.dart is a bit weird.

Probably we should only have defaults in one location.

Comment (jonasfj) [reply app/lib/shared/handler_helpers.dart:136]: But I think we can clean that up later.

File: app/lib/frontend/handlers/headers.dart [hunk 2]
```diff
@@ -13,21 +13,36 @@ class CacheHeaders {
...
+    assert(!private && !public, 'both private and public is not allowed');
+  }
 
   Map<String, String> call() {
     return <String, String>{
       HttpHeaders.cacheControlHeader: <String>[
-        requestContext.isNotAuthenticated
-            ? 'public'
-            : (signedInStorage ?? 'private'),
+        if (private || (!public && requestContext.isNotAuthenticated))
```
Comment (sigurdm) [inline app/lib/frontend/handlers/headers.dart:42]: I find this hard to parse (too many negations).

What happens if neither `private` nor `public` but `requestContext.isAuthenticated` - is that possible? Will it end up public?

Comment (sigurdm) [reply app/lib/frontend/handlers/headers.dart:42]: Maybe something along:
```suggestion
        if (public || requestContext.isNotAuthenticated)
          'public'
        else
          'private',
```

Comment (jonasfj) [reply app/lib/frontend/handlers/headers.dart:42]: If `private` then we force it 'private'
else if `public` then we make it 'public'
else if requestContext.isNotAuthenticated then we make it 'public'
else 'private'

Comment (sigurdm) [reply app/lib/frontend/handlers/headers.dart:42]: That's not what I read. I see the last `else` is `'public'`.

**Comment by isoos** [comment]
I think the upload message change could have gone into a different PR, as now tests are failing on that message change.

**Review by isoos — APPROVED** [review]
